widgets/heatmap/heatmapdemo/heatmapdemo.go

```diff
@@ -24,21 +24,86 @@ import (
 	"github.com/mum4k/termdash/terminal/tcell"
 	"github.com/mum4k/termdash/terminal/terminalapi"
 	"github.com/mum4k/termdash/widgets/heatmap"
+	"math/rand"
+	"time"
 )
 
+// playHeatMap continuously changes the displayed values on the heat map once every delay.
+// Exits when the context expires.
+func playHeatMap(ctx context.Context, hp *heatmap.HeatMap, delay time.Duration) {
+	const max = 100
+
+	ticker := time.NewTicker(delay)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			maxRows, maxCols := hp.ValueCapacity()
+			rows := rand.Intn(maxRows) + 1
+			cols := rand.Intn(maxCols) + 1
+
+			var values [][]float64
+			for i := 0; i < rows; i++ {
+				rv := make([]float64, cols)
+				for j := 0; j < cols; j++ {
+					rv = append(rv, float64(rand.Int31n(max+1)))
+				}
+				values = append(values, rv)
+			}
+
+			if err := hp.Values(nil, nil, values); err != nil {
+				panic(err)
+			}
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func getData() ([]string, []string, [][]float64) {
+	var xl = []string{
+		"one",
+		"two",
+		"three",
+		"four",
+		"five",
+		"six",
+	}
+	var yl = []string{
+		"one",
+		"two",
+		"three",
+		"four",
+		"five",
+	}
+	var values = [][]float64{
+		{1, 2, 3, 4, 5, 6},
+		{-1, -2, -3, -4, -5, -6},
+		{7, 8, 9, 10, 11, 12},
+		{12, 11, 10, 9, 8, 7},
+		{6, 5, 4, 3, 2, 1},
+	}
+	return xl, yl, values
+}
+
 func main() {
 	t, err := tcell.New()
 	if err != nil {
 		panic(err)
 	}
 	defer t.Close()
 
-	hp, err := heatmap.New()
+	hp, err := heatmap.New(heatmap.HideXLabels(), heatmap.HideYLabels())
 	if err != nil {
 		panic(err)
 	}
 
-	// TODO: set heatmap's data
+	if err := hp.Values(getData()); err != nil {
+		panic(err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	//go playHeatMap(ctx, hp, 1*time.Second)
```

Comment:
Additionally when I downsize (resize) the terminal window with the heatmap demo, the application crashes and we get:

```go
panic: container.Draw => error: unable to draw widget *heatmap.HeatMap: unable to set cell runes in area (66,0)-(78,1), it must fit inside the available cell area is (0,0)-(75,22)
```

I perform these tests, because most of the widgets that I have implemented had a problem with this initially. These bugs usually meant that I didn't include the necessary size adjustment or error handling in the Draw method, or that the widget didn't specify the correct options back to the infrastructure. Please feel free to reach our if you will have trouble finding the issue, we can debug it together.

Reply:
Resolved, but I found a new bug. The first column of the canvas can't be displayed:
- should be `PRESS` instead of `RESS`
- only show one digit of the Y labels 
![image](https://user-images.githubusercontent.com/26627380/101726104-d8dd3280-3aec-11eb-9745-c30d762e2aac.png)

Reply:
This one looks interesting. I tried playing with the demo, but was unable to reproduce it. How can we get it into this state?

Also as mentioned before - please feel free to let me know if you would like my help debugging any of these issues. I will gladly cooperate here.

Reply:
I run all demos like this. But when I use `termbox` instead of `tcell` in `heatmapdemo.go`, it becomes normal.

Reply:
Good catch, I managed to reproduce this in other demo and filed #281. Feel free to develop this with `termbox` to avoid the issue. I will work on tracking it down.

Reply:
Fyi #283 fixes the issue.